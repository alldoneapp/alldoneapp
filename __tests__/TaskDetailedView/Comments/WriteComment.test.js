/**
 * @jest-environment jsdom
 */
import * as React from 'react'
import WriteComment from '../../../components/TaskDetailedView/CommentsView/WriteComment'

import renderer from 'react-test-renderer'

describe('WriteComment component', () => {
    describe('WriteComment snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer.create(<WriteComment />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })

    describe('WriteComment methods', () => {
        it('should change the comment text', () => {
            const tree = renderer.create(<WriteComment />)
            const instance = tree.getInstance()
            instance.onChangeText('asd')
            expect(instance.state.commentText).toEqual('asd')
        })

        it('should reset things when on blur', () => {
            const tree = renderer.create(<WriteComment />)
            const instance = tree.getInstance()
            instance.onBlur()
            expect(instance.state.height._value).toEqual(42)
        })

        it('should react properly to multiline content changes', () => {
            const tree = renderer.create(<WriteComment />)
            const instance = tree.getInstance()
            instance.onContentSizeChange({ nativeEvent: { contentSize: { height: 42 } } })
            expect(instance.state.height._value).toEqual(42)
        })

        it('should open the select file dialog on press attachment', () => {
            const tree = renderer.create(<WriteComment />)
            const instance = tree.getInstance()
            instance.onPressAttachment()
            const input = document.getElementsByTagName('input')
            expect(input).toBeTruthy()
        })
    })
})
