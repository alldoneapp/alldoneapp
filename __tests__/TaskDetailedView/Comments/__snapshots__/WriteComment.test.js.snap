// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WriteComment component WriteComment snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 42,
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": 6,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#8A94A6",
            "fontFamily": "alldone",
            "fontSize": 20,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderColor": "#C6CDD2",
        "borderRadius": 4,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "height": 42,
        "justifyContent": "space-between",
        "marginLeft": 16,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onContentSizeChange={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Write an update"
      placeholderTextColor="#8A94A6"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Regular",
            "fontSize": 16,
            "letterSpacing": 0.32,
            "lineHeight": 24,
          },
          Object {
            "flex": 1,
            "paddingLeft": 16,
            "paddingRight": 32,
            "paddingVertical": 8,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "left": "100%",
          "opacity": 1,
          "position": "absolute",
          "top": "25%",
          "transform": Array [
            Object {
              "translateX": -28,
            },
          ],
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#8A94A6",
              "fontFamily": "alldone",
              "fontSize": 22,
              "opacity": 1,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
